PugSharp.Match/Match.cs

```diff
@@ -91,12 +91,16 @@ private void InitializeStateMachine()
             .PermitDynamicIf(MatchCommand.LoadMatch, () => HasRestoredMatch() ? MatchState.RestoreMatch : MatchState.WaitingForPlayersConnectedReady);
 
         _MatchStateMachine.Configure(MatchState.WaitingForPlayersConnectedReady)
-            .PermitDynamicIf(MatchCommand.PlayerReady, () => HasRestoredMatch() ? MatchState.MatchRunning : MatchState.MapVote, AllPlayersAreReady)
+            .PermitDynamicIf(MatchCommand.PlayerReady, () => HasRestoredMatch() ? MatchState.MatchRunning : MatchState.PreMapVote, AllPlayersAreReady)
             .OnEntry(StartWarmup)
             .OnEntry(SetAllPlayersNotReady)
             .OnEntry(StartReadyReminder)
             .OnExit(StopReadyReminder);
 
+        _MatchStateMachine.Configure(MatchState.PreMapVote)
+           .PermitDynamicIf(MatchCommand.MapListCheckCompleted, () => OneMapConfigured() ? MatchState.TeamVote : MatchState.MapVote)
```

Comment:
I think it would be better to do the check and state transition in the state WaitingForPlayersConnectedReady and also split the dynamic permit into two transitions